test_programs/compile_success_empty/field_or_integer_static_trait_method/src/main.nr

```diff
@@ -0,0 +1,21 @@
+trait Read {
+    fn read(data: [Field; 1]) -> Self;
+}
+
+impl Read for Field {
+    fn read(data: [Field; 1]) -> Self {
+        data[0]
+    }
+}
+
+impl Read for u32 {
+    fn read(data: [Field; 1]) -> Self {
+        data[0] as u32
+    }
+}
+
+fn main() {
+    let data = [1];
+    let _: u32 = u32::read(data);
+    let _: Field = Field::read(data);
```

Comment:
Can you add asserts on the return values so that we can ensure the correct version is called?

E.g. each impl could add a different number to the result.